Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -45,46 +45,72 @@ UPSWorldSubsystem& UPSWorldSubsystem::Get(const UObject& WorldContextObject)
 	return *ThisSubsystem;
 }
 
-// Returns current row of progression system
-const FPSRowData& UPSWorldSubsystem::GetCurrentRow() const
-{
-	const UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	return SaveGameInstance->GetCurrentRow();
-}
-
 // Set current row of progression system by tag
 void UPSWorldSubsystem::SetCurrentRowByTag(FPlayerTag NewRowPlayerTag)
 {
-	// check if current row is similar
-	if (GetCurrentRow().Character == NewRowPlayerTag)
+	for (const auto& KeyValue : ProgressionSettingsDataInternal)
 	{
-		return;
-	}
+		const FPSRowData& RowData = KeyValue.Value;
 
-	UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	SaveGameInstance->SetRowByTag(NewRowPlayerTag);
-	OnCurrentRowDataChanged.Broadcast(NewRowPlayerTag);
+		if (RowData.Character == NewRowPlayerTag)
+		{
+			CurrentRowNameInternal = KeyValue.Key;
+			OnCurrentRowDataChanged.Broadcast(NewRowPlayerTag);
+			return; // Exit immediately after finding the match
+		}
+	}
 }
 
-// Returns previous row of progression system
-const FPSRowData& UPSWorldSubsystem::GetPreviousRow() const
+// Returns previous row by current name
+FPSRowData& UPSWorldSubsystem::GetPreviousRow()
 {
-	const UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
-	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
-
-	return SaveGameInstance->GetPreviousRow();
+	return GetPreviousRowByCurrentName();
 }
 
 // Returns the data asset that contains all the assets of Progression System game feature
 const UPSDataAsset* UPSWorldSubsystem::GetPSDataAsset() const
 {
+	ensureMsgf(PSDataAssetInternal, TEXT("ASSERT: [%i] %s:\n'PSDataAssetInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
 	return PSDataAssetInternal.LoadSynchronous();
 }
 
+//  Returns a current save to disk row name
+const FName UPSWorldSubsystem::GetFirstSaveToDiskRowData() const
+{
+	ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %s:\n'SaveGameDataInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
+	return SaveGameDataInternal->GetSavedProgressionRowByIndex(0);
+}
+
+//  Returns a current save to disk row by name
+FPSSaveToDiskData& UPSWorldSubsystem::GetCurrentSaveToDiskRowByName()
+{
+	ensureMsgf(SaveGameDataInternal, TEXT("ASSERT: [%i] %s:\n'SaveGameDataInternal' is empty!"), __LINE__, *FString(__FUNCTION__));
+	return SaveGameDataInternal->GetSaveToDiskDataByName(CurrentRowNameInternal);
+}
+
+// Returns a current progression settings row by name 
+FPSRowData& UPSWorldSubsystem::GetCurrentProgressionSettingsRowByName()
+{
+	return *ProgressionSettingsDataInternal.Find(CurrentRowNameInternal);
```

Comment:
is unsafe because Find() may return a nullptr if CurrentRowNameInternal is not found in the map. Dereferencing a nullptr will lead to undefined behavior and potentially crash the program.

### Expected result:
```cpp
if (const FPSSaveToDiskData* FoundRow = ProgressionSettingsDataInternal.Find(CurrentRowNameInternal))
{
    return *FoundRow;
}
return FPSSaveToDiskData::EmptyData
```

Reply:
I agree, aded: https://github.com/h4rdmol/ProgressionSystem/pull/18/commits/8507ab58e56ba02e83ec823162d804318dabbdd4